src/pages/permissions/panels/IdentityGroupsPanelConfirmModal.tsx

```diff
@@ -0,0 +1,134 @@
+} from "util/permissionIdentities";
+import { useSettings } from "context/useSettings";
+import GroupsOrIdentityChangesTable from "./GroupOrIdentityChangesTable";
+import { updateGroupsForIdentities } from "api/auth-identities";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { useToastNotification } from "context/toastNotificationProvider";
+import usePanelParams from "util/usePanelParams";
+
+interface Props {
+  onConfirm: () => void;
+  close: () => void;
+  selectedIdentities: LxdIdentity[];
+  addedGroups: Set<string>;
+  removedGroups: Set<string>;
+}
+
+const IdentityGroupsPanelConfirmModal: FC<Props> = ({
+  onConfirm,
+  close,
+  addedGroups,
+  removedGroups,
+  selectedIdentities,
+}) => {
+  const { data: settings } = useSettings();
+  const [groupBy, setGroupBy] = useState<"identity" | "group">("identity");
+  // const [submitting, setSubmitting] = useState(false);
+  const notify = useNotify();
+  const panelParams = usePanelParams();
+  const queryClient = useQueryClient();
+  const toastNotify = useToastNotification();
+  const hasLoggedInIdentity = selectedIdentities.some(
+    (identity) => identity.id === settings?.auth_user_name,
+  );
+
+  const identityGroupsChangeSummary = getChangesInGroupsForIdentities({
+    identities: selectedIdentities,
+    addedGroups,
+    removedGroups,
+  });
+
+  const groupIdentitiesChangeSummary = pivotIdentityGroupsChangeSummary(
+    identityGroupsChangeSummary,
+  );
+
+  const handleChangeGroupBy = () => {
+    setGroupBy((prev) => {
+      if (prev === "identity") {
+        return "group";
+      }
+
+      return "identity";
+    });
+  };
+
+  const handleSaveGroupsForIdentities = () => {
+    // setSubmitting(true);
+
+    const newGroupsForIdentities = generateGroupAllocationsForIdentities({
+      addedGroups,
+      removedGroups,
+      identities: selectedIdentities,
+    });
+
+    updateGroupsForIdentities(selectedIdentities, newGroupsForIdentities)
+      .then(() => {
+        // modifying groups should invalidate both identities and groups api queries
+        void queryClient.invalidateQueries({
+          predicate: (query) => {
+            return [queryKeys.identities, queryKeys.authGroups].includes(
+              query.queryKey[0] as string,
+            );
+          },
+        });
+        const successMessage =
+          selectedIdentities.length > 1
+            ? `Updated groups for ${selectedIdentities.length} identtiies`
+            : `Updated groups for ${selectedIdentities[0].name}`;
+        toastNotify.success(successMessage);
+        panelParams.clear();
+      })
+      .catch((e) => {
+        notify.failure("Update groups failed", e);
+      })
+      .finally(() => {
+        // setSubmitting(false);
+        onConfirm();
+      });
+  };
+
+  // TODO: make upstream change to include loading state for ConfirmationModal
+  return (
+    <ConfirmationModal
```

Comment:
The modal has double scrollbars if the changes overflow in height.

![image](https://github.com/canonical/lxd-ui/assets/1155472/9c25b17d-bb54-415d-be90-227330c95814)